apps/web-giddh/src/app/gst/gst-setting/gst-setting.component.ts

```diff
@@ -0,0 +1,339 @@
+                    this.toaster.showSnackBar('success', this.responseArray[response.lutIndex].successMessage);
+                }
+                this.changeDetection.detectChanges();
+            }
+        });
+    }
+
+    /**
+     * This will be use for init lut form
+     *
+     * @memberof GstSettingComponent
+     */
+    public initGstSettingForm(): void {
+        this.gstSettingForm = this.formBuilder.group({
+            gstData: this.formBuilder.array([
+                this.initLutForm()
+            ])
+        });
+        this.paymentIntegrateForm = this.formBuilder.group({
+            withPay: [false]
+        });
+    }
+
+    /**
+     *This will be use for lut form FormArray group
+     *
+     * @return {*}  {FormGroup}
+     * @memberof GstSettingComponent
+     */
+    public initLutForm(data?: any): FormGroup {
+        return this.formBuilder.group({
+            fromDate: [data?.fromDate ?? ''],
+            toDate: [data?.toDate ?? ''],
+            lutNumber: [data?.lutNumber ?? ''],
+            uniqueName: [data?.uniqueName ?? '']
+        });
+    }
+
+    /**
+     * Get label for export type
+     *
+     * @returns {string}
+     * @memberof GstSettingComponent
+     */
+    public getExportTypeLabel(): string {
+        if (this.activeCompany) {
+            let value = this.activeCompany.withPay ? this.commonLocaleData.app_yes.toLowerCase() : this.commonLocaleData.app_no.toLowerCase();
+            const exportType = this.exportTypes.find(item => item?.value === value);
+            return exportType ? exportType.label : '';
+        }
+    }
+
+    /**
+    * Saves export type
+    *
+    * @memberof GstSettingComponent
+    */
+    public setExportType(event?: any): void {
+        if (event && event.value && this.exportType !== event.value) {
+            this.paymentIntegrateForm.get('withPay')?.patchValue(event.value === this.commonLocaleData.app_yes.toLowerCase());
+            const value = event.value === this.commonLocaleData.app_yes.toLowerCase() ? true : false;
+            this.store.dispatch(this.settingsProfileActions.PatchProfile({ withPay: value }));
+        }
+    }
+
+    /**
+    * This will use for get lut item list
+    *
+    * @memberof GstSettingComponent
+    */
+    public getLutList(): void {
+        this.componentStore.getLutNumberList();
+    }
+
+    /**
+     * This will be use for add new lut item
+     *
+     * @param {*} [user]
+     * @memberof GstSettingComponent
+     */
+    public addNewLutItem(user?: any): void {
+        let mappings = this.gstSettingForm.get('gstData') as FormArray;
+        if (user) {
+            mappings.push(this.initLutForm({
+                fromDate: this.fromDate,
+                toDate: this.toDate
+            }));
+            let lastIndex = mappings.controls.findIndex(control => !control.get('lutNumber')?.value);
+            let lastFormArray = mappings.at(lastIndex);
+            let fromDate = null;
```

Comment:
move the fromDate and toDate variable initialization and value assignment in the below if condition